logic/concludable/ConjunctionConcludable.java

```diff
@@ -112,21 +119,86 @@ public Value asValue() {
     }
 
+    Optional<Map<Reference, Set<Reference>>> updatedUnifier(
+            Variable conjVar, Variable headVar, Map<Reference, Set<Reference>> variableMapping) {
+        if (!hintsIntersect(headVar, conjVar)) return Optional.empty();
+        Map<Reference, Set<Reference>> cloneOfMapping = cloneMapping(variableMapping);
+        cloneOfMapping.putIfAbsent(conjVar.reference(), new HashSet<>());
+        cloneOfMapping.get(conjVar.reference()).add(headVar.reference());
+        return Optional.of(cloneOfMapping);
+    }
+
+    Map<Reference, Set<Reference>> cloneMapping(Map<Reference, Set<Reference>> mapping) {
+        Map<Reference, Set<Reference>> clone = new HashMap<>();
+        mapping.forEach((key, set) -> clone.put(key, new HashSet<>(set)));
+        return clone;
+    }
+
     public static class Relation extends ConjunctionConcludable<RelationConstraint, ConjunctionConcludable.Relation> {
 
         public Relation(final RelationConstraint constraint) {
             super(copyConstraint(constraint));
         }
 
         @Override
-        public Stream<Unification> unify(HeadConcludable.Relation unifyWith) {
-            // Check the relation variables' isa constraint labels and prune if there is no intersection
-
-            // Find all roleplayer mapping combinations, which should have the form:
-            // Set<List<Pair<RelationConstraint.RolePlayer, RelationConstraint.RolePlayer>>> rolePlayerMappings
-            // For each, prune if there is no label intersection (or any other pruning, e.g. by value)
-            // Then build a Unification for each valid combination
-            return Stream.empty(); // TODO
+        public Stream<Map<Reference, Set<Reference>>> unify(HeadConcludable.Relation unifyWith) {
+            if (this.constraint().players().size() > unifyWith.constraint().players().size()) return Stream.empty();
+            Map<Reference, Set<Reference>> startingUnifier = new HashMap<>();
+            Optional<Map<Reference, Set<Reference>>> newUnifier;
+            if (constraint.owner().reference().isName()) {
+                newUnifier = updatedUnifier(this.constraint().owner(), unifyWith.constraint().owner(), startingUnifier);
+                if (newUnifier.isPresent()) startingUnifier = newUnifier.get();
+                else return Stream.empty();
+            }
+
+            if (constraint.owner().isa().isPresent() && constraint.owner().isa().get().type().reference().isName()) {
+                assert unifyWith.constraint().owner().isa().isPresent();
+                newUnifier = updatedUnifier(this.constraint().owner().isa().get().type(),
+                        unifyWith.constraint().owner().isa().get().type(), startingUnifier);
+                if (newUnifier.isPresent()) startingUnifier = newUnifier.get();
+                else return Stream.empty();
+            }
+
+            return matchRolesAndUnify(new ArrayList<>(this.constraint().players()), unifyWith.constraint().players(),
```

Comment:
this should be called `unifyRolePlayers` and you should just use the name `unifier` instead of `startingUnifier`. Check out my later comment on some ideas for this